### ISSUE TITLE
Add documentation for VM dev machine(using free or open source software)

### ISSUE THREAD

**Issue by CorbanR** [firecracker-microvm/firecracker#1033]
Some great documentation has been added via https://github.com/firecracker-microvm/firecracker/pull/940. However the current documentation is referencing paid software (such as VMware). It would be nice to figure out an alternative free option when setting up local dev.

**Comment by gj** [comment]
I'm all for adding documentation for alternatives, but just as a note the steps in #940 were done using the free "evaluation" copy of Fusion 🙂

**Comment by CorbanR** [comment]
You eventually have to buy Fusion if you want to continue to use it yes?

**Comment by gj** [comment]
It's possible to continue cycling the Fusion trial in perpetuity, but it's certainly not an ideal setup. I think this issue is really valuable and definitely didn't mean to suggest otherwise. I just wanted to point out that if "free" is the main qualifier, it's possible to achieve that with Fusion. That being said, not relying on a loophole in a trial version of a paid, closed source product definitely sounds like a better path forward 😄

**Comment by raduweiss** [comment]
I think the free&open dev setup is probably QEMU-based. If anyone wants to contribute instructions for that it would be great.

**Comment by karuppiah7890** [comment]
@raduweiss I recently was able to run with VirtualBox in MacOS. Shall I put something together to tell how to run it and may be we can see if that could be added as a doc? I also used `vagrant` to manage the VM

**Comment by raduweiss** [comment]
Yeah, that would be useful. Feel free to open a PR against https://github.com/firecracker-microvm/firecracker/blob/master/docs/dev-machine-setup.md .

**Comment by karuppiah7890** [comment]
Okay! 😄

**Comment by richardcase** [comment]
Some of the engineers at WW are using Vagrant (and Virtualbox) to build a Linux machine for running Firecracker when using MacOS.

**Comment by richardcase** [comment]
@jmickey - your vagrant work could be adapted to help with this issues. wdyt?

**Comment by JonathanWoollett-Light** [comment]
We are open to further contribution to update the documentation here if anyone has an alternative.

**Comment by bchalios** [comment]
Yes, feel free to give it ago @ambersongliu. Thanks for the interest.

**Comment by ambersongliu** [comment]
Hi all, we were thinking about proposing VirtualBox as the free alternative to VMWare. Here are some notes that I've taken about it. I'll work on the documentation of how to download VirtualBox with your go-ahead, or I can explore another technology! 

VirtualBox is a cross-platform nested virtualization software that runs on Intel-based macOS and all modern Windows hosts. The core package is licensed under GNU GPL v2 and exposes VT-X/AMD-V extensions to guests to run KVM-style hypervisors (such as Firecracker) inside the VM.

Benefits of VirtualBox:

-Cross-platform support
    -It is supported on 64-bit Windows and macOS (Sierra to Mojave) 
- Open source and free for commercial use
- Nested virtualization support (VT-x/AMD-V)
- Can integrate with tools, such as Vagrant, for provisioning

**Comment by richardcase** [comment]
> we were thinking about proposing VirtualBox as the free alternative to VMWare.

Some of the people in a team that i worked in used VirtualBox when working on Firecracker related things.  They generally used vagrant to create the VM: https://github.com/liquidmetal-dev/flintlock/blob/main/Vagrantfile
